services:
  db:
    image: postgres:16
    container_name: adman-db
    restart: unless-stopped
    volumes:
      - adman_postgres_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_DB=adman
      - POSTGRES_USER=adman
      - POSTGRES_PASSWORD=SuperHesloProAdmanDb_sK5!g9@p
    networks:
      - internal
  app:
    container_name: adman-app
    restart: unless-stopped
    build: .
    user: "root"
    volumes:
      - ./app:/app # FINÁLNÍ OPRAVA: Mapujeme pouze adresář s Django aplikací
      - /srv/clients:/srv/clients
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - internal
      - proxy
    depends_on:
      - db
    environment:
      - DATABASE_URL=postgres://adman:SuperHesloProAdmanDb_sK5!g9@p@db:5432/adman
      - SECRET_KEY=NahodnyTajnyKlicProDjango_aB1$$c2!fG
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.adman.rule=Host(`adman.it-enterprise.solutions`)"
      - "traefik.http.routers.adman.entrypoints=websecure"
      - "traefik.http.routers.adman.tls.certresolver=letsencrypt"
      - "traefik.http.services.adman.loadbalancer.server.port=8000"
      - "traefik.docker.network=proxy"
  
  filebrowser:
    image: filebrowser/filebrowser:latest
    container_name: adman-filebrowser
    restart: unless-stopped
    user: "root"
    volumes:
      - /srv/clients:/srv
      - ./filebrowser_db:/database
    networks:
      - proxy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.adman-files.rule=Host(`files.adman.it-enterprise.solutions`)"
      - "traefik.http.routers.adman-files.entrypoints=websecure"
      - "traefik.http.routers.adman-files.tls.certresolver=letsencrypt"
      - "traefik.http.services.adman-files.loadbalancer.server.port=80"
      - "traefik.docker.network=proxy"

volumes:
  adman_postgres_data:
networks:
  proxy:
    external: true
  internal:
    driver: bridge